launcher/minecraft/mod/ModFolderModel.h

```diff
@@ -85,8 +90,20 @@ class ModFolderModel : public ResourceFolderModel {
 
     bool isValid();
 
+    bool setResourceEnabled(const QModelIndexList& indexes, EnableAction action) override;
+    QModelIndexList getAffectedMods(const QModelIndexList& indexes, EnableAction action);
+
     RESOURCE_HELPERS(Mod)
 
+   public:
+    QStringList requiresList(QString id);
+    QStringList requiredByList(QString id);
+
    private slots:
     void onParseSucceeded(int ticket, QString resource_id) override;
+    void onParseFinished();
+
+   private:
+    QHash<QString, QList<Mod*>> m_requiredBy;
+    QHash<QString, QList<Mod*>> m_requires;
```

Comment:
Can you not get away with doing this to eliminate duplicates?
```suggestion
    QHash<QString, QSet<Mod*>> m_requiredBy;
    QHash<QString, QSet<Mod*>> m_requires;
```